src/components/dashboard-navbar.js

```diff
@@ -103,55 +121,109 @@ export const DashboardNavbar = (props) => {
-        ],
-        signature: signature,
-        signerAddress: address
-      };
-      checkAndCreateUser(newUser).catch((error) => {
-        console.error(error)
-      })
-      dispatch(userActions.userAccount(address));
-      dispatch(userActions.connecion());
-      localStorage.setItem("userAddress", address);
+  const handleClickConnectWalletMenu = (event) => {
+    connectWalletSetAnchorEl(event.currentTarget);
+  };
+  const handleCloseConnectWalletMenu = () => {
+    connectWalletSetAnchorEl(null);
+  };
+
+  const connectEthereumWallet = async () => {
+    {
+      let provider;
+      let library;
+      let accounts;
+      let network;
+      try {
+        provider = await getWalletProvider().connect();
+        library = new ethers.providers.Web3Provider(provider);
+        accounts = await library.listAccounts();
+        network = await library.getNetwork();
+      } catch (error) {
+        console.error(error);
+      }
+      setProvider(provider);
+      setLibrary(library);
+      setNetwork(network);
+      setSelectedBlockchain("Ethereum")
+
+      console.log(network);
+
+      if (accounts) {
+        const address = accounts[0];
+        const signature = await signMessage(library)
+          .catch((error) => {
+            console.error(error)
+          });
+        console.log(new Date().getTime())
+        const newUser = {
+          wallets: [
+            {
+              name: "Default Account",
+              blockchain: selectedBlockchain, // TODO: enable other login accounts, for now only Ethereum account
+              chainId: network.chainId,
+              address: address,
+              createdAt: new Date().getTime()
+            }
+          ],
+          signature: signature,
+          signerAddress: address
+        };
+        checkAndCreateUser(newUser).catch((error) => {
+          console.error(error)
+        })
+        dispatch(userActions.userAccount(address));
+        dispatch(userActions.connecion());
+        dispatch(userActions.blockchain("Ethereum"));
+        localStorage.setItem("userAddress", address);
+        localStorage.setItem("blockchain", "Ethereum");
+      }
+    };
+  }
+
+  const connectSolanaWallet = async () => {
+    if (connected) {
+      disconnectSolWallet()
     }
-  }; 
+    setSelectedBlockchain("Solana")
+    setOpenSolanaWalletDialog(true)
+  }
+  const completeSolanaWalletConnect = async () => {
+    // console.log(publicKey.toBase58());
+    //sign 
+    const message = ["Log in CrossPay", userAddress];
+    const encodedMessage = new TextEncoder().encode(message);
+    const signature = await solWalletSignMessage(encodedMessage)
 
+    const newUser = {
+      wallets: [
+        {
+          name: "Default Account",
+          blockchain: selectedBlockchain,
+          chainId: process.env.REACT_APP_CLUSTER,
+          address: publicKey.toBase58(),
+          createdAt: new Date().getTime()
+        }
+      ],
+      signature: signature,
```

Comment:
convert it to String
We are sending buffer. 
![image](https://user-images.githubusercontent.com/39519012/215120151-fe618ec0-1cf5-41a5-b196-148325d60a2c.png)

Reply:
fixed